XCode 8

Hi, I tried today to add this library to my project using Carthage with XCode 8 Beta 6 (also using Swift 3) and got this linker error:

"error: The following binaries use incompatible versions of Swift:"

Could this be a Carthage bug, or could it be a ZivArchive one?

Thanks

I tried, but got this error:

`$ carthage bootstrap --no-use-binaries
*** Checking out ZipArchive at "v1.5"
*** xcodebuild output can be found in /var/folders/_f/rpjhxdls3p966hqhkt0pd7h00000gp/T/carthage-xcodebuild.0anOS9.log
*** Building scheme "ZipArchive-Mac" in ZipArchive.xcodeproj
*** Building scheme "ZipArchive-iOS" in ZipArchive.xcodeproj
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
A shell task (/usr/bin/xcrun xcodebuild -project /Users/julian/projects/Harvard/Wyss/Square/v1.0/src/Carthage/Checkouts/ZipArchive/ZipArchive.xcodeproj -scheme ZipArchive-iOS -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** CLEAN FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
`

I will start building with xCode 8 once released. But that doesn't help now.

> Julián da Silva Gillig <mailto:notifications@github.com>
> Thursday, August 25, 2016 7:00 AM
>
> I tried, but got this error:
>
> `$ carthage bootstrap --no-use-binaries
> *** Checking out ZipArchive at "v1.5"
> *** xcodebuild output can be found in 
> /var/folders/_f/rpjhxdls3p966hqhkt0pd7h00000gp/T/carthage-xcodebuild.0anOS9.log
> *** Building scheme "ZipArchive-Mac" in ZipArchive.xcodeproj
> *** Building scheme "ZipArchive-iOS" in ZipArchive.xcodeproj
> ** CLEAN FAILED **
>
> The following build commands failed:
> Check dependencies
> (1 failure)
> ** BUILD FAILED **
>
> The following build commands failed:
> Check dependencies
> (1 failure)
> A shell task (/usr/bin/xcrun xcodebuild -project 
> /Users/julian/projects/Harvard/Wyss/Square/v1.0/src/Carthage/Checkouts/ZipArchive/ZipArchive.xcodeproj 
> -scheme ZipArchive-iOS -configuration Release -sdk iphoneos 
> ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode 
> CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) 
> failed with exit code 65:
> ** CLEAN FAILED **
>
> The following build commands failed:
> Check dependencies
> (1 failure)
> ** BUILD FAILED **
>
> The following build commands failed:
> Check dependencies
> (1 failure)
> `
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/ZipArchive/ZipArchive/issues/276#issuecomment-242396968>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAq37uUvcSlJbnDR25tl2kT_1Oo-b6-_ks5qjZ_1gaJpZM4Jn6b9>.
>
> Julián da Silva Gillig <mailto:notifications@github.com>
> Thursday, August 18, 2016 2:03 PM
>
> Hi, I tried today to add this library to my project using Carthage 
> with XCode 8 Beta 6 (also using Swift 3) and got this linker error:
>
> "error: The following binaries use incompatible versions of Swift:"
>
> Could this be a Carthage bug, or could it be a ZivArchive one?
>
> Thanks
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/ZipArchive/ZipArchive/issues/276>, or mute the 
> thread 
> <https://github.com/notifications/unsubscribe-auth/AAq37nEMWnPsK12iYHxKdsvUsouFeO2Uks5qhMiSgaJpZM4Jn6b9>.
>

--- a/SSZipArchive/SSZipArchive+Swift.swift
+++ b/SSZipArchive/SSZipArchive+Swift.swift
@@ -10,12 +10,12 @@ import Foundation
 
 extension SSZipArchive {
     
-    static func unzipFileAtPath(path: String, toDestination destination: String, overwrite: Bool, password: String?, delegate: SSZipArchiveDelegate?) throws -> Bool {
+    static func unzipFileAtPath(_ path: String, toDestination destination: String, overwrite: Bool, password: String?, delegate: SSZipArchiveDelegate?) throws -> Bool {
         
         var success = false
         var error: NSError?
         
-        success = __unzipFileAtPath(path, toDestination: destination, overwrite: overwrite, password: password, error: &error, delegate: delegate)
+        success = __unzipFile(atPath: path, toDestination: destination, overwrite: overwrite, password: password, error: &error, delegate: delegate)
         if let throwableError = error {
             throw throwableError
         }

--- a/ZipArchive.xcodeproj/project.pbxproj
+++ b/ZipArchive.xcodeproj/project.pbxproj
@@ -328,6 +328,7 @@
 					};
 					B423AE191C0DF76A0004A2F1 = {
 						CreatedOnToolsVersion = 7.1.1;
+						LastSwiftMigration = 0800;
 					};
 				};
 			};
@@ -561,6 +562,7 @@
 				PRODUCT_NAME = "$(PROJECT_NAME)";
 				SKIP_INSTALL = YES;
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 3.0;
 			};
 			name = Debug;
 		};
@@ -582,6 +584,7 @@
 				PRODUCT_BUNDLE_IDENTIFIER = com.samsoffes.ZipArchive;
 				PRODUCT_NAME = "$(PROJECT_NAME)";
 				SKIP_INSTALL = YES;
+				SWIFT_VERSION = 3.0;
 			};
 			name = Release;
 		};

